src/state.rs

```diff
@@ -1,16 +1,32 @@
 use crate::providers::Providers;
+use crate::relay::RelayClient;
 use crate::store::ClientStore;
 use crate::{env::Config, providers::ProviderKind};
 use build_info::BuildInfo;
 use opentelemetry::metrics::{Counter, UpDownCounter};
 use opentelemetry::sdk::trace::Tracer;
+use std::sync::Arc;
 use tracing_subscriber::prelude::*;
 
+#[derive(Clone)]
 pub struct Metrics {
     pub registered_webhooks: UpDownCounter<i64>,
     pub received_notifications: Counter<u64>,
 }
 
+pub trait State<S>
+where
+    S: ClientStore,
+{
+    fn get_config(&self) -> Config;
```

Comment:
The naming convention is to omit `get_` prefix for getters, and have only the `set_` prefix for setters.